cogeo_mosaic/backends/dynamodb.py

```diff
@@ -122,25 +167,32 @@ def write(self, overwrite: bool = False, **kwargs: Any):
                 )
             self.delete()
 
-        items = self._create_items()
-        self._write_items(items)
+        # Validation and Create a new instance
+        if not isinstance(mosaic, MosaicJSON):
+            mosaic = MosaicJSON(**dict(mosaic))
+        else:
+            mosaic = deepcopy(mosaic)
 
-    def _update_quadkey(self, quadkey: str, dataset: List[str]):
-        """Update single quadkey in DynamoDB."""
-        self.table.put_item(
-            Item={"mosaicId": self.mosaic_name, "quadkey": quadkey, "assets": dataset}
+        items: List[Dict[str, Any]] = []
+
+        # Create Metadata item
+        meta = json.loads(self.mosaic_def.json(exclude={"tiles"}), parse_float=Decimal)
+        items.append(
+            {"quadkey": self._metadata_quadkey, "mosaicId": self.mosaic_name, **meta}
         )
 
-    def _update_metadata(self):
-        """Update bounds and center.
+        # Create Tile items
+        for quadkey, assets in self.mosaic_def.tiles.items():
+            items.append(
+                {"mosaicId": self.mosaic_name, "quadkey": quadkey, "assets": assets}
+            )
 
-        Note: `parse_float=Decimal` is required because DynamoDB requires all numbers to be in Decimal type
```

Comment:
Nit: I think it would be good to keep this comment around somewhere. I was surprised to need to use `Decimal`.